### PR TITLE
Page with Bookmarks 🌟

### PR DESCRIPTION
**Pull request by frederikebruendermann**
✨The focus of this UserStory was to **create a Page with all Bookmarks** so that it´s **possible to see all the bookmarked concerts** ✨

What did I do in this PullRequest?

- Created HomePage where all the bookmarked concerts are shown
- Implement React Router
- Created Nav Bar component for the Pages
- Connect Pages through NavBar
- Smaller changes on the design & testing

[Have a look at ConcertLife on Vercel 👀](https://capstone-project-d0rujfysr-frederikebruendermann.vercel.app/)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -20,25 +24,38 @@ export default function App() {
       .catch(error => console.error(error))
   }, [currentCity])
 
+  console.log(bookmarks)
+  console.log(concerts)
```
Comment (LisanneArdern) [inline src/App.js]: I would delete console.logs

File: src/App.js [hunk 2]
```diff
@@ -20,25 +24,38 @@ export default function App() {
...
-      {activePage === 'suche' && (
-        <SearchPage
-          pageName="Suche"
-          concerts={concerts}
-          onNavigate={handleClickDetails}
-          onSubmit={chosenCity}
-        />
-      )}
+      <Switch>
+        <Route exact path="/suche">
```
Comment (LisanneArdern) [inline src/App.js]: you don't need exact path here (or on the other pages)

```suggestion
        <Route  path="/suche">
```

File: src/App.js [hunk 3]
```diff
@@ -20,25 +24,38 @@ export default function App() {
...
-      )}
+        <Route exact path="/details">
+          <DetailPage
+            pageName="Details"
+            concert={concertDetails}
+            onNavigate={handleClickBack}
+            onBookmark={handleBookmark}
+          />
+        </Route>
+        <Route exact path="/">
```
Comment (LisanneArdern) [inline src/App.js]: This is the only time you need exact path

File: src/components/NavBar.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import styled from 'styled-components/macro'
+import PropTypes from 'prop-types'
+
+NavBar.propTypes = {
+  onClickSearch: PropTypes.func,
+  onClickHome: PropTypes.func,
+}
+
+export default function NavBar({ onClickSearch, onClickHome }) {
```
Comment (LisanneArdern) [inline src/components/NavBar.js]: Yesterday I received feeback for better naming: 
```suggestion
export default function NavBar({ onSearch, onHome }) {
```

File: src/components/NavBar.spec.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { render } from '@testing-library/react'
+import NavBar from './NavBar'
+
+describe('it calls onClickSearch when clicking on the navbar', () => {
+  it('renders', () => {
+    const onClickSearch = jest.fn()
+
+    render(<NavBar onClick={onClickSearch} />)
+  })
```
Comment (LisanneArdern) [inline src/components/NavBar.spec.js]: I would specify the test more and test if onClickSearch is called when clicking on the Search button and onClickHome when clicking on the Home button

File: src/components/NavBar.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import styled from 'styled-components/macro'
+import PropTypes from 'prop-types'
+
+NavBar.propTypes = {
+  onClickSearch: PropTypes.func,
+  onClickHome: PropTypes.func,
```
Comment (LisanneArdern) [inline src/components/NavBar.js]: not required?

File: src/components/NavBar.stories.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+    import NavBar from './NavBar'
+    export default {
+        title: 'NavBar',
+        component: NavBar
+    }
+      
+    const Template = args => <NavBar {...args} />
+      
+    export const Default = Template.bind({})
+    Default.args = {}
```
Comment (LisanneArdern) [inline src/components/NavBar.stories.js]: Storybook becomes powerful when showing different options:)

File: src/pages/HomePage.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+  return (
+    <Wrapper>
+      <Heading pageName={pageName} />
+      <Hint>
+        <BookmarkInactive>☆</BookmarkInactive>
+        <h2>Merkliste</h2>
+      </Hint>
+      <div>
+        {bookmarks.length === 0 ? (
+          <Text>Du hast Dir noch keine Konzerte gemerkt.</Text>
```
Comment (LisanneArdern) [inline src/pages/HomePage.js]: Very nice!

File: src/pages/HomePage.js [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+  display: flex;
+  align-items: center;
+  gap: 10px;
+`
+
+const Wrapper = styled.section`
+  padding: 10px;
+`
+
+const ConcertList = styled.ul`
```
Comment (LisanneArdern) [inline src/pages/HomePage.js]: The Concert list is a ul but then the ConcertCard is a section. I am not sure if this is sematic html

File: src/pages/HomePage.spec.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+            id: 1,
+            artist: 'Thees',
+            date: 'Fr., 02.07.2021, 20:00',
+            location: 'Bremen, Seebühne Bremen',
+          },
+        ]}
+        onClick={onNavigate}
+      />
+    )
+    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument()
```
Comment (LisanneArdern) [inline src/pages/HomePage.spec.js]: what you are testing here is that a h1 is rendered which is not described in the test

File: src/pages/HomePage.stories.js [hunk 11]
```diff
@@ -0,0 +1,31 @@
+import HomePage from './HomePage'
```
Comment (LisanneArdern) [inline src/pages/HomePage.stories.js]: Brilliant Storybook file!

**Review by LisanneArdern — COMMENTED** [review]
Really good work! 🎸 
Just a few remarks :)

File: src/pages/HomePage.js [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+  onNavigate: PropTypes.func,
+  bookmarks: PropTypes.array,
+}
+
+export default function HomePage({ bookmarks, pageName, onNavigate }) {
+  return (
+    <div>
+      <Heading pageName={pageName} />
+      <ConcertList>
+        {bookmarks.length === 0 ? (
```
Comment (Dajanajozanovic) [inline src/pages/HomePage.js]: You could shorten the code by writing it like this: {bookmaks.length > 0 ? instead of "==="

File: src/App.js [hunk 13]
```diff
@@ -60,11 +65,19 @@ export default function App() {
 
   function handleClickDetails(id) {
     setConcertId(id)
-    setActivePage('details')
+    history.push('/details')
```
Comment (Dajanajozanovic) [inline src/App.js:65]: you could also use history.goBack --> it will be bring you back to the side where you've been before. (if it's interesting for you)

**Review by Dajanajozanovic — COMMENTED** [review]
Good job! I just added some small recommendations. 👍 👯‍♀️

**Pull request merged**